src/debugger/appWorker.ts

```diff
@@ -203,6 +206,9 @@ export class MultipleLifetimesAppWorker {
             if (object.method === "prepareJSRuntime") {
                 // The MultipleLifetimesAppWorker will handle prepareJSRuntime aka create new lifetime
                 this.gotPrepareJSRuntime(object);
+            } else if (object.method === "$disconnected") {
+                // We need to shutdown the current app worker, and create a new lifetime
+                this.singleLifetimeWorker = null;
```

Comment:
This doesn't actually do anything to the worker; do we need to actively stop it (e.g. what if there's a timer running in the worker)? And what about the websocket connection, should we close that, or we wait for the other party to end it?